File: src/algor/MainKDTNearest.java
package algor;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 * Created by tieorange on 18/03/2017.
 */
public class MainKDTNearest {

    public static void main(String args[]) throws IOException {
//        BufferedReader in = new BufferedReader(new FileReader("input.txt"));
        int numpoints = 5;

        KDTree kdTree = new KDTree(numpoints);
        double x[] = new double[2];

        double[][] matrxi = new double[2][2]; // MATRIX

        x[0] = 2.1;
        x[1] = 4.3;
        kdTree.add(x);

        x[0] = 3.3;
        x[1] = 1.5;
        kdTree.add(x);

        x[0] = 4.7;
        x[1] = 11.1;
        kdTree.add(x);

        x[0] = 5.0;
        x[1] = 12.3;
        kdTree.add(x);

        x[0] = 5.1;
        x[1] = 1.2;
        kdTree.add(x);

        System.out
                .println("Enter the co-ordinates of the point: (one after the other)");
        InputStreamReader reader = new InputStreamReader(System.in);
        BufferedReader br = new BufferedReader(reader);
        double sx = Double.parseDouble(br.readLine());
        double sy = Double.parseDouble(br.readLine());

        double userInputArray[] = {sx, sy};
        KDNode kdNode = kdTree.find_nearest(userInputArray);
        System.out.println("The nearest neighbor is: ");
        System.out.println("(" + kdNode.x[0] + " , " + kdNode.x[1] + ")");
//        in.close();
    }
}

File: src/algor/MainKDTNearest.kt
package algor

import algor.FileReader.COLUMNS_COUNT
import algor.FileReader.getDataArray
import algor.FileReader.splitDataRow
import java.io.IOException

/**
 * Created by tieorange on 18/03/2017.
 */
object MainKDTNearest {

    private val trainingData: List<DoubleArray> = getDataArray("train.txt")

    private val testDataArray: List<DoubleArray> = getDataArray("test.txt")

    @Throws(IOException::class)
    @JvmStatic fun main(args: Array<String>) {
        val kdTree = KDTree(trainingData)

        performTestData(testDataArray, kdTree)

        askUserForInput(kdTree)
    }

    private fun askUserForInput(kdTree: KDTree) {
        print("Type in data (for ex. 5.7,2.5,5.0,2.0,Iris-virginica): ")
        val line = readLine()
        if (line != null) {
//            COLUMNS_COUNT = 5
            val splitDataRow = splitDataRow(line)
//            COLUMNS_COUNT = 4
            performTestData(listOf(splitDataRow), kdTree)
        }
    }

    private fun performTestData(testDataArray: List<DoubleArray>, kdTree: KDTree) {
        var correctAnswers = 0
        for (testDataRow in testDataArray) {
            val kdNode = kdTree.find_nearest(testDataRow)
            val isCorrect = kdNode?.printNeighbor(testDataRow) ?: false
            if (isCorrect) correctAnswers++
        }

        val accuracyRate = (correctAnswers / testDataArray.size.toFloat()) * 100.0
        println("accuracyRate = $accuracyRate")
    }

    fun getUserInput() {
        /* println("Enter the co-ordinates of the point: (one after the other)")
         val reader = InputStreamReader(System.`in`)
         val br = BufferedReader(reader)
         val xUserInput = br.readLine().toDouble()
         val yUserInput = br.readLine().toDouble()*/
    }

}